File: 2187.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int teste = 1;

        while (true) {
            int V = sc.nextInt();
            if (V == 0) break;

            int I = V / 50; V %= 50;
            int J = V / 10; V %= 10;
            int K = V / 5;  V %= 5;
            int L = V;

            System.out.println("Teste " + teste++);
            System.out.println(I + " " + J + " " + K + " " + L);
            System.out.println();
        }

        sc.close();
    }
}

File: 2187.c
#include <stdio.h>

int main() {
    int V;
    int teste = 1;

    while (1) {
        scanf("%d", &V);
        if (V == 0) break;

        int I = V / 50;
        V %= 50;

        int J = V / 10;
        V %= 10;

        int K = V / 5;
        V %= 5;

        int L = V;

        printf("Teste %d\n", teste++);
        printf("%d %d %d %d\n\n", I, J, K, L);
    }

    return 0;
}
